R/pokehelperr.R

```diff
@@ -20,7 +23,33 @@
 #' get_types(list('Pikachu', 'Eevee', 'Charizard', ...))
 #'
 get_types <- function(pokemon_names) {
-  # Function code (TBD in Milestone 3)
+   url <- "https://raw.githubusercontent.com/UBC-MDS/pokehelpyer/main/data/pokemon.csv"
+    names_types_df  <- read.csv(url)
+
+    names_types_df <- names_types_df |>
+        mutate(Name = trim(Name), Name = tolower(Name)) 
+
+    name_with_symbol <- str_extract(names_types_df$Name, ".*[^\\w\\s].*")
+    name_with_symbol <- name_with_symbol[!is.na(name_with_symbol)]
+
+    poke_types <- list()
+    for (name in pokemon_names){
+        name <- trim(name)
+        name <- tolower(name)
+
+        if (!(name %in% name_with_symbol)){
+            name <- str_remove(name, "[^a-zA-Z0-9 ]")
+        }
+
+        type <- names_types_df |> 
+            filter(Name == name) |>
+            select(`Type.1`, `Type.2`) 
```

Comment:
Whenever you use functions from another package, you need to namespace them, e.g. `dplyr::filter` and `dplyr::select`

Reply:
See https://r-pkgs.org/whole-game.html#use_package.